File: 백준/Bronze/30642. 아이그루스와 화장실/아이그루스와 화장실.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        String mascot = br.readLine();
        int K = Integer.parseInt(br.readLine());

        if (mascot.equals("induck")) {
            if (K % 2 == 0) {
                System.out.println(K);
            } else {
                if (K + 1 > N) {
                    System.out.println(K - 1);
                } else {
                    System.out.println(K + 1);
                }
            }
        } else {
            if (K % 2 == 1) {
                System.out.println(K);
            } else {
                if (K + 1 > N) {
                    System.out.println(K - 1);
                } else {
                    System.out.println(K + 1);
                }
            }
        }

        br.close();
    }
}

File: 백준/Bronze/30642. 아이그루스와 화장실/아이그루스와 화장실.c
#include <stdio.h>

int main() {
    int N, K;
    char mascot[8];

    scanf("%d", &N);
    scanf("%s", mascot);
    scanf("%d", &K);

    if (strcmp(mascot, "induck") == 0) {
        if (K % 2 == 0) {
            printf("%d\n", K);
        } else {
            if (K + 1 > N) {
                printf("%d\n", K - 1);
            } else {
                printf("%d\n", K + 1);
            }
        }
    } else {
        if (K % 2 == 1) {
            printf("%d\n", K);
        } else {
            if (K + 1 > N) {
                printf("%d\n", K - 1);
            } else {
                printf("%d\n", K + 1);
            }
        }
    }

    return 0;
}
